### PR TITLE
Remove Travis CI and Appveyor configurations

### PR DESCRIPTION
**Pull request by mdickinson**
We've been using GitHub Actions successfully for a while now, and Travis CI has been increasingly problematic.

This PR removes the Travis CI and Appveyor configurations, leaving only the GitHub actions.

---

EDIT: This PR also removes the restriction on installing NumPy on Python 3.9.  That restriction was only there for the benefit of Travis CI - installing NumPy on the GitHub Actions platform seems to work fine.

---

EDIT:  This PR additionally removes _all_ Python 3.9 restrictions!

### CODE REVIEW

**Comment by mdickinson** [comment]
I've changed the Appveyor configuration (through the Appveyor web page) to "ignore branches without appveyor.yml".  Closing and re-opening to make sure I get an all-green CI run.

File: setup.py [hunk 1]
```diff
@@ -312,13 +312,11 @@ def get_long_description():
             "Sphinx>=2.1.0,!=3.2.0",
             # Python 3.9 exclusions:
             #
-            # * NumPy installation fails on Python 3.9 on the Ubuntu Xenial
-            #   system used by Travis CI.
-            # * PySide2 is installable but also currently not working
+            # * PySide2 is installable but currently not working
```
Comment (rahulporuri) [inline setup.py:315]: this is tangential but i just tried creating a python 3.9 venv and i was able to successfully install pyside2 with traits/traitsui/pyface and run the traits testsuite.

**Review by rahulporuri — APPROVED** [review]
LGTM

File: setup.py (hunk 1 above)
Comment (mdickinson) [reply setup.py:315]: Ah, nice. I'll give that a try in this PR, since we're already messing with the NumPy part.

Comment (mdickinson) [reply setup.py]: > I'll give that a try in this PR, since we're already messing with the NumPy part.

And it works! @rahulporuri still okay to merge?

**Review by rahulporuri — APPROVED** [review]
still LGTM

File: setup.py (hunk 1 above)
Comment (kitchoi) [reply setup.py]: Just curious, is it because a newer version pyside2 is made available to work on Python 3.9?

Comment (mdickinson) [reply setup.py]: @kitchoi Not 100% sure, but I think the last time I was trying to make this work it was with a Python 3.9 pre-release, so working PySide2 wheels didn't exist on PyPI. Or possibly the PyPI wheels simply weren't compatible with Travis CI's flavour of Linux, but are compatible with the GitHub Actions flavour.

Comment (kitchoi) [reply setup.py]: That's good to know. Thank you!

**Pull request merged**